game/game.go

```diff
@@ -548,8 +582,39 @@ func (g *Game) Leavable() bool {
 	return !g.Started
 }
 
-func (g *Game) Joinable() bool {
-	return !g.Closed && g.NMembers < len(variants.Variants[g.Variant].Nations) && len(g.ActiveBans) == 0 && len(g.FailedRequirements) == 0
+func (g *Game) IsGameMasterInvited(email string) bool {
+	for _, invitation := range g.GameMasterInvitations {
+		if invitation.Email == email {
+			return true
+		}
+	}
+	return false
+}
+
+func (g *Game) HasReplaceableMember() bool {
+	for _, member := range g.Members {
+		if member.Replaceable {
+			return true
+		}
+	}
+	return false
+}
+
+func (g *Game) Joinable(user *auth.User) bool {
+	if len(g.ActiveBans) > 0 || len(g.FailedRequirements) > 0 {
+		return false
+	}
+	if g.Closed || g.NMembers >= len(variants.Variants[g.Variant].Nations) {
+		if g.GameMasterEnabled && g.HasReplaceableMember() && (!g.RequireGameMasterInvitation || g.IsGameMasterInvited(user.Email)) {
```

Comment:
Can a player be "replaceable" if there's no game master?

Reply:
I guess it's the beginning of more complex features, but it wreaks havoc (as you have mentioned) on the stats, so I'd prefer the havoc to be contained to private games, which game master games are at the moment.

WDYT?

Reply:
That's great, just wanted to check you'd thought about it

Reply:
👍